src/api/Interpret.h

```diff
@@ -64,6 +64,55 @@ class DefinedFunctions {
     }
 };
 
+class TermNames {
+public:
+    bool has(std::string const & name) const { return nameToTerm.find(name) != nameToTerm.end(); }
+    bool has(PTRef term) const { return termToNames.find(term) != termToNames.end(); }
+
+    void insert(std::string const & name, PTRef term, bool scoped = false) {
+        assert(not has(name));
+        nameToTerm[name] = term;
```

Comment:
```suggestion
        nameToTerm.emplace(name, term);
```
is slightly more efficient

Reply:
Updated!